Icemoon

Whoneedspacee opened the pull request:
## About The Pull Request

Adds a couple new ruins and a new megafauna to be the equivalent of blood drunk miner on ice moon. Adds a capability for weather to not target indoor areas, removing the need for an area target on a z level. Areas can now specify if they want megafauna or mobs or tunnels to go into them. There are now new spawners for 2 new mobs on ice moon wolves and polar bears. Tunnels can now have varying widths you can define on tunnel producing type. Ice moon also has a unique planetary atmos like lavaland. Admins can pick which mining map they want to be picked next, otherwise it will be chosen randomly. Ruins can now spawn in multiple area types on the same z level which fixes an issue with multiz ruins. The ice moon budget for spawning ruins is larger due to the multiz making things very spread out.

https://m.youtube.com/watch?v=iB-hkN_wYmY&t=
https://m.youtube.com/watch?v=iB-hkN_wYmY&t=
https://m.youtube.com/watch?v=iB-hkN_wYmY&t=

## Why It's Good For The Game

Variation is fun, this provides a lot of tools to make other additions like this possible as well. This isn't the most finished thing I've put out in terms of new content but I'm hoping the multiz will inspire people to make unique ruins, and I'll probably add some myself over time.

## Changelog
:cl:
add: Ice moon, sometimes spawns in the place of lavaland with a couple different ruins, mobs, and a lower z level
admin: Admins can choose the next mining map in the same tab as normal map choosing
/:cl:

Whoneedspacee committed:
there was an attempt

Whoneedspacee committed:
adds ice moon map

adds config options for choosing mining maps

areas now have options that affect random generation on them

weather now has an option to not target indoor areas

adds base for icemoon ruins

turns many 1's into booleans

adds ice chasms

adds ztraits for weather stuff

Whoneedspacee committed:
fixes up ice ruins to be ice moon compatible

adds lower z level to the ice moon mine

ice chasms now have a smoothed icon

megafauna only spawn in the underground portion of ice moon now

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
openspace fixes

Whoneedspacee committed:
adds new areas and underground specific ruins

KomradeSpectre committed:
Adds the abandoned village ruin for Ice Moon

Whoneedspacee committed:
Merge pull request #1 from KomradeSpectre/icemoon

Adds the abandoned village ruin for Ice Moon

Whoneedspacee committed:
adds abandoned village ruin and fixes some area and tile stuff to work with ice moon

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
random mining maps are no longer picked in the config

you can no longer change the mining map before setup is complete

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
adds above and below ground ruin as a test

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
adds debug functionality for ice moon ruins

fixes a bug where multiz was impossible unless there were shared areas because of whitelisted areas

adds multiple whitelist areas for ruin placement in a list

all underground areas are now outdoors

underground plasma lava rivers are now only spawned in unexplored areas so ruins don't get destroyed by them

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
adds unique spawners to icemoon to replace tendrils

adds 2 new mobs to icemoon the polar bear and wolf

adds a clothing flag for shoes that dont slip on ice

modifies mining site to be ready except for the boss

adds the ability for tunnel width to be a specification, ice moon can have 1 tile wide tunnels and rarely 2 wide tunnels

adds a no caves mineral for ice moon ruins

Whoneedspacee committed:
wolves no longer run into lava or chasms

bears now enrage sooner

bear spawners can now actually be created

adds base for ice moon atmospherics

adds base for a new boss and achievements / score

really bad coder icon for ice boots added

Whoneedspacee committed:
ice moon now has it's own planetary atmos! (thank god)

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
new frost miner stuff

megafauna recovery time can now be tailored to have different cooldowns to ranged or melee attacks
by default sets both of them, with two arguments the first is ranged cooldown and the second is melee cooldown

Whoneedspacee committed:
converts a bunch of lavaland maps to be compatible with ice moon

adds tendrils from lavaland to the bottom of ice moon with the other megafauna

tralezab commented:
thats the way uh huh uh huh i like it

itseasytosee commented:
Does it have any gib ability? It seems like it would be really easy to cheese with dead bodies

Whoneedspacee commented:
At the end it gibs the target, works the same as any other megafauna.

Shadowflame909 commented:
How do we get the ash-drakes white, and shooting out whatever chemical is in that syndicate Freon grenade. Glyon? No clue, but the ice grenade of liquid death.

Whoneedspacee commented:
give me sprites and i will do great things with them

the snowballs are just the nuclear particle sprite this is what im working with

cacogen commented:
It uses multi-z? So it's going to be buggy as fuck.

Farquaar commented:
Looks cool! How does this affect the gulag?

trollbreeder commented:
@Shadowflame909 it's called frost oil i think, it rapidly cools you down, it's also a very potent lizard exterminator

necromanceranne commented:
Nice, I didn't know you were making it multiz! That's gonna be sweet!

Whoneedspacee commented:
> It uses multi-z? So it's going to be buggy as fuck.

so we're going to fix the bugs really good 👍

Dennok commented:
> It uses multi-z? So it's going to be buggy as fuck.

I think it means multi-z ruins, this different thing now. It create ruin room on special z-lvl with ladder to small ruin entrance on mining area.

Dorsisdwarf commented:
Oh damn, this looks incredible! Lavaland use of multiZ might be a better platform for testing than multistations anyway

ATH1909 commented:
Finally, we can make chasms not insta-kill people (we can just make them send people down a z-level instead).

lolman360 commented:
good pr.

are we gonna get something akin to tendrils here or is there going to be another way to get infinite mobs/special loot?

itseasytosee commented:
Why do the wolves drop corgi skin and meat?

Dennok commented:
> good pr.
> 
> are we gonna get something akin to tendrils here or is there going to be another way to get infinite mobs/special loot?

`cave entrance` is Icemoon spawner that can spawns wolfs and bears now.

Whoneedspacee commented:
> Why do the wolves drop corgi skin and meat?

thanks for pointing out things that were placeholder 👍

bobbahbrown commented:
Marked WIP per discussion with Whoneedsspace on 2020-02-17.

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

bobbahbrown commented on code/modules/mapping/ruins.dm:
```diff
@@ -11,8 +11,15 @@
+					if(istype(new_area, type)) // it's at least one of our types so it's whitelisted
+						valid = TRUE
+						break
+			if(!valid)
```
you could move the valid = FALSE assignment to outside the loop no need to reassign each time with the break

Whoneedspacee committed:
updates mob drops for ice moon mobs

updates config entry default value for ice moon

updates ice moon map to have new gulag

Dennok commented:
Maybe make special wall around boss that you show in video?
Special hard rock that can be mined by mastered miner and immune to boss attacks. 
I hate this megafauna like colossus that anihilate all nearby place when someone fight it.

IndieanaJones commented:
I'm happy to see this is finally here.  It'll take a bit to get the Ice Moon to be as content rich as Lavaland, but I think that'll be interesting in its own right.  Also, if the Ice Moon is selected, does it change the planet in the background visible from the station?

Whoneedspacee commented:
thats one of the few things i need to change as well as the shuttle sendoff text in the ui there thanks for reminding me

if anyone has that parallax or knows how the hell that shit works please dm me or post here, i know someone made it before

IndieanaJones commented:
Also, a couple of personal thoughts:
- Polar bears should have something more than just being tendril-less goliaths.  Perhaps they could have a pounce akin to the xeno hunter, where you can to step out of the way in order to not get tackled?
- The miner megafauna should probably not be a miner.  As it stands, the Blood-Drunk miner is sort of a lazy and forgettable entity who barely qualifies as a megafauna, he'd be much more interesting and cool if he was some sort of monster instead
- Ash Walkers should have some equal on the ice moon, but probably not be the same.  Other ghost roles like golems, syndies, and podpeeps can fit in both places, but the ashies are heavily linked to lavaland.  Could literally be the same with a race change, or something to change down the line.

Whoneedspacee commented:
1. sure

2. ice moon was meant to have an eldritchy background so this is kinda that with the miner being possessed or something, if you have sprites to improve it sure, but i took a sprite and made a boss from it

3. sprites

Dennok commented:
3. You can use cat peoples

FantasticFwoosh commented:
Very talented attempt, i've had eyes on a Ice-Moon concept for a while I hope you can take it to new places rather than turn the heat down on Lavaland. Contact me through DM if you need any sprite requests as i have a little cache of concept doodles and time at the moment.

>     1. You can use cat girls
Im not sure they need the encouragement of diversification + resulting drama about seeking approval to do it. Begrudgingly recommending bear people or satirical comedy human russians maybe.

trollbreeder commented:
territorial russians that run on 19-20th century technology and use a rickidy-dickidy ol' fabricator and they're chaotic neutral, yes i love the idea already

fludd12 commented:
catgirls totally meet the requirement for eldritch theme, though
they can worship...

...

*nya*~rlathothep. 😎

PKPenguin321 commented:
I think @Qustinnus wrote parallax for us? Or at least the lavaland planet part

Whoneedspacee commented:
if i could have it my way then ice moon would be comprised entirely of new content but if i did that then it would never get merged

if anyone wants to add ruins to this feel free, you can dm me if you want assistance or items added for it

Farquaar commented:
There's an spear-wielding eskimo sprite in simplehuman.dmi that could probably be used as a placeholder for the miner sprite until someone makes a good looking monster.

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

Whoneedspacee committed:
updates station parallax for icemoon

removes extra lavaland ported ruins

updates to demonic frost miner

adds ladder to icemoon map near station

Valtosin commented:
I testmerged this PR, so, there is some issues with it:

1. You can't build/dig ladder/stairs if you fall down. Only one ladder at center.
2. You also can't dig down.
3. Ashwalkers isn't cold-resistant people. They die without boots, even if wintercoat.
3.1 Also campfire doesn't work as heat source.
4. Wolfs and bears are strong.
5. You can fall into ruin's end from above and loot all things without completing it.

Whoneedspacee committed:
updates ice moon map to have the ladder inside the mining base

plant flora dont spawn on the lower z level of ice moon now

you get sand from mining rocks now

buncha demonic frost miner changes holy cow

Whoneedspacee committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into icemoon

optimumtact commented:
I don't want to carry all of these sprites, code and content for something that shows up every other round.

I'm not a fan of lavaland as it is, as I think it needs to be more firewalled from the station, and there's too much of a focus from people on content that isn't the main station.

This could be a main station map, or a gamemode, but I don't see us adding this as a lavaland "reskin"

FancyCyclops commented:
Oshan Lab.
